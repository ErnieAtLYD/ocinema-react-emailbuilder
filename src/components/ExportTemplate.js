// @flow
import React, { useState } from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import { inlineContent } from 'juice';
import Inky from 'react-inky';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import LayoutTemplateWrapper from './LayoutTemplateWrapper';
import parse from 'siphon-media-query';

import newsletterStyles from '../styles/newsletter.scss';
import './ExportTemplate.scss';

// This gets called from the exportAsHTML method in /actions
export const getExportedHTML = (layout: any): any => {
  const markup = renderToStaticMarkup(
    <Inky>
      <Inky.Head>
        <style>
          {parse(newsletterStyles)}
        </style>
      </Inky.Head>
      <Inky.Body>
        {layout.map((item: NewsletterLayoutItemType, index: any): React$Element<any> => <LayoutTemplateWrapper item={item} />)}
      </Inky.Body>
    </Inky>
  );
  return inlineContent(Inky.doctype + markup, newsletterStyles);
}

type ComponentType = {
  layout: Array<any>,
  exportAsHTML: Function
}

const ExportTemplate = ({ layout, exportAsHTML }: ComponentType): React$Element<({children?: React$Node}) => React$Node> => {
  const [open, setOpen] = useState(false);
  const [newsletterHTML, setHTML] = useState();

  const handleOpen = (): void => {
    setOpen(true);
  };
  const handleClose = (): void => {
    setOpen(false);
  };
  const handleExport = (layout: any): void => {
    exportAsHTML();
    setHTML(getExportedHTML(layout));
    handleOpen();
  }

  return (
    <>
      <Button onClick={(): void => { handleExport(layout) }}>
        Export as HTML
      </Button>
      <Modal
        open={open}
        onClose={handleClose}>
        <div className="modal">
          <p>Copy and paste the HTML.</p>
          <textarea
            value={newsletterHTML}
          >
          </textarea>
          <Button>
            Copy to Clipboard
          </Button>
        </div>
      </Modal>
    </>
  )
}

export default ExportTemplate;
