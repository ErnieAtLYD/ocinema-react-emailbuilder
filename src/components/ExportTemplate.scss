.modal {
  position: absolute;
  width: 40%;
  background-color: #fff;
  border: 2px solid #000;
  padding: 10px;

  textarea {
    width: 100%;
    height: 300px;
  }
}
