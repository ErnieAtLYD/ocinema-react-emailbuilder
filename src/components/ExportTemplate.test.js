import React from 'react';
import { shallow } from 'enzyme';
import ExportTemplate from './ExportTemplate'

describe('ExportTemplate', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ExportTemplate  />);
    // console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });
});
