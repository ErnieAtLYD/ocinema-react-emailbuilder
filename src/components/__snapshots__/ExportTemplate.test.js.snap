// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportTemplate renders without crashing 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
  >
    Export as HTML
  </WithStyles(ForwardRef(Button))>
  <ForwardRef(Modal)
    onClose={[Function]}
    open={false}
  >
    <div
      className="modal"
    >
      <p>
        Copy and paste the HTML.
      </p>
      <textarea />
      <WithStyles(ForwardRef(Button))>
        Copy to Clipboard
      </WithStyles(ForwardRef(Button))>
    </div>
  </ForwardRef(Modal)>
</Fragment>
`;
